docs/examples/segmenting_search.md

```diff
@@ -0,0 +1,96 @@
+# Segmenting Search Queries
+
+In this example, we will demonstrate how to leverage the `MultiTask` and `enum` features of OpenAI Function Call to segment search queries. We will define the necessary schemas using Zod and demonstrate how segment queries into multiple sub queries and execute them in parallel.
+
+!!! tips "Motivation"
+
+    Extracting a list of tasks from text is a common use case for leveraging language models. This pattern can be applied to various applications, such as virtual assistants like Siri or Alexa, where understanding user intent and breaking down requests into actionable tasks is crucial. In this example, we will demonstrate how to use OpenAI Function Call to segment search queries and execute them in parallel.
+
+## Structure of the Data
+
+`SearchTypeSchema` is a Zod schema that defines the structure of a search query object. It has three fields: `title`, `query`, and `type`. The `title` field is the title of the request, the `query` field is the query to search for relevant content, and the `type` field is the type of search. The `executeSearch` function is used to execute the search query.
+
+```ts
+import Instructor from "@/instructor"
+import OpenAI from "openai"
+import { z } from "zod"
+
+const oai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY ?? undefined,
+  organization: process.env.OPENAI_ORG_ID ?? undefined
+})
+
+const client = Instructor({
+  client: oai,
+  mode: "FUNCTIONS"
+})
+
+const SearchTypeSchema = z
+  .enum(["VIDEO", "EMAIL"])
+  .describe("Enumeration representing the types of searchs that can be performed")
+
+const SearchSchema = z
+  .object({
+    title: z.string().describe("Title of the request"),
+    query: z.string().describe("Query to search fro relevant content"),
+    type: SearchTypeSchema.describe("Type of search")
```

Comment:
since this is being defined independently with its own description, and you are adding a description here - I wonder what the actual JSON schema output is - tbh I'm not sure - does this double up the description on this one property? does the second supersede the first? You probably don't need 2, though, to describe this same property, right?

Reply:
Yeah, I would just replace it with type:  z.enum.